cc_corpus/deduplication.py

```diff
@@ -100,35 +102,65 @@ def __del__(self):
         self.close()
 
 
-def read_batch(batch_file_prefix):
+def read_batch_to_memory(batch_file_prefix: Path):
```

Comment:
Please pull out the common part of `read_batch` and `read_batch_to_memory` into a separate (generator) function (e.g. `_read_batch`). It could yield the doc_ids, minhashes and the name of the doc file.